src/services/userService.ts

```diff
@@ -11,28 +12,30 @@ export const upsertUser = (
   containerName: NonEmptyString,
   blobName: NonEmptyString
 ) =>
-  getBlobData(blobServiceClient, containerName, blobName, UsersCompanies)
-    .chain(users =>
-      fromPredicate(
-        (usersCompanies: UsersCompanies) =>
-          usersCompanies.find(
-            u => u.fiscalCode === userCompanies.fiscalCode
-          ) !== undefined,
-        _ => [..._, userCompanies]
-      )(users).foldTaskEither<Error, UsersCompanies>(
-        _ => taskEither.of(_),
-        _ =>
-          taskEither.of([
-            ..._.filter(e => e.fiscalCode !== userCompanies.fiscalCode),
-            userCompanies
-          ])
+  pipe(
+    getBlobData(blobServiceClient, containerName, blobName, UsersCompanies),
+    TE.chain(
+      flow(
+        TE.fromPredicate(
+          (usersCompanies: UsersCompanies) =>
+            usersCompanies.find(
+              u => u.fiscalCode === userCompanies.fiscalCode
+            ) !== undefined,
+          _ => [..._, userCompanies]
```

Comment:
I guess you can use `filterOrElse`, something like:

```suggestion
    getBlobData(blobServiceClient, containerName, blobName, UsersCompanies),
    TE.filterOrElse(
          flow(A.findFirst(_ => _.fiscalCode === userCompanies.fiscalCode), O.isSome),
          _ => [..._, userCompanies]
```